frontend/app/components/menu.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { Link } from '@remix-run/react';
+
+import { useTranslation } from 'react-i18next';
+
+interface MenuItem {
+  id: string;
+  key: string;
+  value: string;
+  path: string;
+  showIcon: boolean;
+}
+interface MenuProps {
+  menuList: MenuItem[];
+}
+export function Menu({ menuList }: MenuProps) {
+  const { t } = useTranslation('gcweb');
+  const [showDropdown, setShowDropdown] = useState<boolean>(false);
+  const dropdown = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    // Hide dropdown when click outside
+    if (!showDropdown) return;
+    function handleClick(event: MouseEvent) {
+      if (dropdown.current && !dropdown.current.contains(event.target as Node)) {
```

Comment:
It's very difficult to understand what this boolean is testing with a quick glance. My suggestion would be to help future developers out by assigning it to a well-named variable.

``` typescript
const targetInsideDropdown = dropdown.current?.contains(event.target as Node) ?? false;
setShowDropdown(!targetInsideDropdown);
```

I *think* I got the nullish fallback right there.. if not you might have to fiddle with it a bit.